packetery/src/Packetery/Order/class-metabox.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Class Metabox
+ *
+ * @package Packetery\Order
+ */
+
+declare( strict_types=1 );
+
+
+namespace Packetery\Order;
+
+use Packetery\Helper;
+use Packetery\Message_Manager;
+
+/**
+ * Class Metabox
+ *
+ * @package Packetery\Order
+ */
+class Metabox {
+
+	/**
+	 * Latte engine.
+	 *
+	 * @var \Latte\Engine
+	 */
+	private $latte_engine;
+
+	/**
+	 * Message manager.
+	 *
+	 * @var Message_Manager
+	 */
+	private $message_manager;
+
+	/**
+	 * Helper.
+	 *
+	 * @var Helper
+	 */
+	private $helper;
+
+	/**
+	 * Order form.
+	 *
+	 * @var \Nette\Forms\Form
+	 */
+	private $order_form;
+
+	/**
+	 * Metabox constructor.
+	 *
+	 * @param \Latte\Engine   $latte_engine Latte engine.
+	 * @param Message_Manager $message_manager Message manager.
+	 * @param Helper          $helper Helper.
+	 */
+	public function __construct( \Latte\Engine $latte_engine, Message_Manager $message_manager, Helper $helper ) {
+		$this->latte_engine    = $latte_engine;
+		$this->message_manager = $message_manager;
+		$this->helper          = $helper;
+	}
+
+	/**
+	 *  Registers related hooks.
+	 */
+	public function register(): void {
+		$this->order_form = new \Nette\Forms\Form();
+		$this->add_fields();
+		add_action( 'add_meta_boxes', array( $this, 'add_meta_boxes' ) );
+		add_action( 'save_post', array( $this, 'save_fields' ) );
+	}
+
+	/**
+	 *  Add metaboxes
+	 */
+	public function add_meta_boxes(): void {
+		add_meta_box(
+			'packetery_metabox',
+			__( 'Packeta', 'packetery' ),
+			array(
+				$this,
+				'render_metabox',
+			),
+			'shop_order',
+			'side',
+			'core'
+		);
+	}
+
+	/**
+	 *  Adds packetery fields to order form.
+	 */
+	public function add_fields(): void {
+		$this->order_form->addHidden( 'packetery_order_metabox_nonce' );
+		$this->order_form->addText( 'packetery_weight', __( 'Weight (kg)', 'packetery' ) )
```

Comment:
Zkusil bych pro přehlednost na ten addText udělat metodu.

Reply:
To se u Nette formulářů nedělá. Novou metodu do třídy typu Form dáváš, když přidáváš nový typ inputu. Např.: ->addDynamic( ... )

https://github.com/Kdyby/FormsReplicator/blob/master/docs/en/index.md